### PR TITLE
Make interval button disabled if user doesn't have access to change it

### PR DESCRIPTION
**Pull request by HelenaEksler**
#261

### CODE REVIEW

**Comment by HelenaEksler** [comment]
User has only permission to view build:
selection_038

User has permissions to update build:
selection_037

**Comment by HelenaEksler** [comment]
@amitaibu Is this the right way to check it?

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 1]
```diff
@@ -36,10 +36,31 @@ public function publicFieldsInfo() {
       'property' => 'field_ci_build_interval',
     );
 
+    $public_fields['can_be_changed'] = array(
+      'property' => 'nid',
+      'process_callbacks' => array(
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: it can just be a `callback`, no?

**Comment by HelenaEksler** [comment]
@amitaibu Ready.

**Comment by amitaibu** [comment]
Needs a re-roll

**Comment by HelenaEksler** [comment]
@amitaibu Done

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 2]
```diff
@@ -48,6 +53,20 @@ public function publicFieldsInfo() {
   }
 
   /**
+   * Callback. Check user has permissions to edit node.
+   *
+   * @param array $value
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: seems wrong.

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 3]
```diff
@@ -48,6 +53,20 @@ public function publicFieldsInfo() {
...
+   *   The image array.
+   *
+   * @return bool
+   *   Returns TRUE if user can edit node.
+   */
+  protected function checkUpdatePermissions($value) {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: $value => $nid

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 4]
```diff
@@ -49,6 +54,21 @@ public function publicFieldsInfo() {
...
+   *
+   * @param EntityMetadataWrapper $wrapper
+   *  CI Build node wrapper.
+   *
+   * @return bool
+   *   Returns TRUE if user can edit node.
+   */
+  protected function checkUpdatePermissions(\EntityMetadataWrapper $wrapper) {
+    $account = $this->getAccount();
+    $node = node_load($wrapper->getIdentifier());
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: $node = $wrapper->value();

$wrapper->access()/ $wrapper->entityAccess()

**Comment by HelenaEksler** [comment]
@amitaibu Ready.

**Pull request merged**